docs: Update README and bump version to 1.4.2

shenxianpeng opened the pull request:
I assume to bump the version to 1.4.2 because

1. The current version in setup.py is incorrect, it is 1.3.1
2. update README based on suggestion in https://github.com/shenxianpeng/cpp-linter-action/issues/55#issuecomment-1106666257

shenxianpeng commented:
Tested looks good 
```bash
python3 -m pip install git+https://github.com/shenxianpeng/cpp-linter-action@v1.4.2

gitpod /workspace/cpp-linter-action (update-readme) $ pip3 show cpp-linter
Name: cpp-linter
Version: 1.4.2
Summary: Bootstrapper for docker's ENTRYPOINT executable.
Home-page: https://github.com/2bndy5/cpp-linter-action
Author: Brendan Doherty
Author-email: 2bndy5@gmail.com
License: MIT
Location: /home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages
Requires: pyyaml, requests
Required-by: 
```

2bndy5 commented:
I'm curious if using `@v1` will allow us to only require future changes when bumping the major version.

[According to the github docs](https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-release-management-for-actions), using `v1` can be shorthand for `v1.x.x`. I think this means we would have to keep moving a separate tag named `v1` to the latest released commit (which is currently tagged with `v1.4.2`). This would be easier than updating the README for every release.

2bndy5 commented:
> I think this means we would have to keep moving a separate tag named v1 to the latest released commit

Or we could create a branch named `v1` and keep pushing deployment-ready commits to that branch (likely cherry-picked from master). But this seem like a bit more work than moving a tag around.

2bndy5 commented:
> The current version in setup.py is incorrect, it is 1.3.1

Ideally we wouldn't need to keep updating this in setup.py if we use the `setuptools_scm` lib as a build dependency. However, I neglected to do this since it adds to the action setup time ([`setuptools_scm`](https://pypi.org/project/setuptools-scm/) is only available via pip).

shenxianpeng commented:
From the tests you run, `v1` makes a lot of sense, and it also works with `python3 -m pip install git+https://github.com/shenxianpeng/cpp-linter-action@v1`

If every release needs to build and upload a wheel to be published to pypi seems to add a bit more work.

shenxianpeng commented on setup.py:
```diff
@@ -17,14 +15,17 @@
+        "A python package that powers the github action named cpp-linter-action. "
+        + f"See `the github repository README <{REPO}#readme>`_ for full details."
+    ),
+    author="Brendan Doherty, Peter Shen",
```
```suggestion
    author="Brendan Doherty",
```

2bndy5 replied:
You don't want your name on this?! I didn't add you email to `author_email` field.

shenxianpeng replied:
You are the author of the Python code, I may be a future contributor, shame, but thank you!

2bndy5 replied:
I consider feedback, review, and testing as contributions.. You deserve it!

shenxianpeng reviewed (commented):
LGTM

2bndy5 commented:
>If every release needs to build and upload a wheel to be published to pypi seems to add a bit more work.

I can easily automate this with CD workflow, but I'm not sure it is worth it right now. Maybe in the future when we have patched in support for other CI services...

shenxianpeng commented:
To save setup time or add a CI workflow for this? Yes, maybe in the future ...

2bndy5 reviewed (approved):
Since I pushed the `v1` tag (and tested it on the test repo with success), I pushed a commit that updated the README to use `v1` instead of the full tag `v1.4.2`.

2bndy5 commented:
> To save setup time

Recent tests (on the test repo) show that pip installing from the repo directly (instead of pypi) only takes about 12 seconds. So, I'm not sure how much time it would actually save (I'm guessing around 5 seconds).

2bndy5 commented:
If you're ready to merge, I can move around the tags afterward (and remove the "pre-release" flag).

shenxianpeng commented:
5 seconds sounds good! OK, I. am going to merge

2bndy5 commented:
FYI, I had to
1. delete the tags
2. push the changes
3. re-create the tags
4. push the changes again

I think this is why the pre-release description of v1.4.2 got removed. BTW, having the tags on a branch that was not the master branch proved to be an additional obstacle.

shenxianpeng commented:
I did

* deleted the pre-release 
* recreate a release v1.4.2 using the tag you created.

I added v1.4.2 to the update-readme branch for testing, it really cause some inconvenience.

2bndy5 commented:
oh! In the future, I'll leave the releases in your capable hands.

shenxianpeng commented:
No problem, I am a release engineer in my other capacity 😆
